src/ProxyClient/HttpDispatcher.php

```diff
@@ -257,19 +275,21 @@ private function setServers(array $servers)
      * Set application base URI that will be prefixed to relative purge and
      * refresh requests, and validate it.
      *
-     * @param string $uriString Your application’s base URI
+     * @param array $baseUris Your application’s base URIs
      *
      * @throws InvalidUrlException If the base URI is not a valid URI
      */
-    private function setBaseUri($uriString = null)
+    private function setBaseUris(array $baseUris = [])
     {
-        if (!$uriString) {
-            $this->baseUri = null;
+        if (0 === \count($baseUris)) {
+            $this->baseUris = [];
 
             return;
         }
 
-        $this->baseUri = $this->filterUri($uriString);
+        foreach ($baseUris as $baseUri) {
+            $this->baseUris[] =  $this->filterUri($baseUri);
```

Comment:
does this detect when $baseUri is empty? i think its good if we produce a clear error message in that case.

Reply:
It cannot be empty. Empty baseUri's were ignored silently up until now and they still are.